src/qt/sendcoinsdialog.cpp

```diff
@@ -828,3 +829,33 @@ void SendCoinsDialog::coinControlUpdateLabels()
         ui->labelCoinControlInsuffFunds->hide();
     }
 }
+
+SendConfirmationDialog::SendConfirmationDialog(const QString &title, const QString &text,
+    int msecDelay, QWidget *parent) :
+    QMessageBox(QMessageBox::Question, title, text, QMessageBox::Yes | QMessageBox::Cancel, parent),
+    msecDelay(msecDelay)
+{
+    setDefaultButton(QMessageBox::Cancel);
+}
+
+int SendConfirmationDialog::exec()
+{
+    button(QMessageBox::Yes)->setEnabled(false);
+    QTimer::singleShot(msecDelay, this, SLOT(enableSendButton()));
```

Comment:
What happens if you close the QMessageBox before the QTimer fires?
I have tested it and it works here... but wondering if it would be better to explicit invalidate the timer in the destructor.